srcpkgs/dkms/files/kernel.d/dkms.postinst

```diff
@@ -40,49 +40,79 @@ for _mod_ in /var/lib/dkms/*; do
 	done
 done
 
+# This sections builds and install the available modules.
+#
+# If either building or installing a module fails, a warning is
+# printed and it is skipped, but the script still tries to build
+# the other modules.
+#
+# The list of available modules is in the form
+# [module1, modulever1, module2, modulever2, ...]
+#
 set -- ${DKMS_MODULES}
 while [ $# -ne 0 ]; do
 	module="$1"
-	modulever="$2"
+	shift
+
+	[ $# -gt 0 ] || break
```

Comment:
I don't think we should reach this condition, right? Two shifts per iteration and an even number of arguments should mean that at one point the number of arguments will reach zero, and the while simply won't run.

Reply:
Reworked to avoid the second test while still verifying that the number of arguments is large enough to avoid a shift error.